packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -479,121 +587,121 @@ export class RunningSummarizer implements IDisposable {
-            return undefined;
-        }
-
-        const pendingTimeoutP = this.pendingAckTimer.start().catch(() => undefined);
-        const summary = this.summaryWatcher.watchSummary(summaryData.clientSequenceNumber);
-
-        // Wait for broadcast
-        const summaryOp = await Promise.race([summary.waitBroadcast(), pendingTimeoutP]);
-        if (!checkNotTimeout(summaryOp)) {
-            return undefined;
-        }
-        this.heuristics.lastAttempted.summarySequenceNumber = summaryOp.sequenceNumber;
-        this.logger.sendTelemetryEvent({
-            eventName: "SummaryOp",
-            timeWaiting: Date.now() - this.heuristics.lastAttempted.summaryTime,
-            refSequenceNumber: summaryOp.referenceSequenceNumber,
-            summarySequenceNumber: summaryOp.sequenceNumber,
-            handle: summaryOp.contents.handle,
-        });
-
-        // Wait for ack/nack
-        const ackNack = await Promise.race([summary.waitAckNack(), pendingTimeoutP]);
-        if (!checkNotTimeout(ackNack)) {
-            return undefined;
-        }
-        this.logger.sendTelemetryEvent({
-            eventName: ackNack.type === MessageType.SummaryAck ? "SummaryAck" : "SummaryNack",
-            category: ackNack.type === MessageType.SummaryAck ? "generic" : "error",
-            timeWaiting: Date.now() - this.heuristics.lastAttempted.summaryTime,
-            sequenceNumber: ackNack.sequenceNumber,
-            summarySequenceNumber: ackNack.contents.summaryProposal.summarySequenceNumber,
-            error: ackNack.type === MessageType.SummaryNack ? ackNack.contents.errorMessage : undefined,
-            handle: ackNack.type === MessageType.SummaryAck ? ackNack.contents.handle : undefined,
-        });
-
-        this.pendingAckTimer.clear();
-
-        // Update for success
-        if (ackNack.type === MessageType.SummaryAck) {
-            this.heuristics.ackLastSent();
-            return true;
-        } else {
-            this.raiseSummarizingError("SummaryNack");
-            return false;
-        }
-    }
-
-    private async generateSummaryWithLogging(message: string, safe: boolean): Promise<GenerateSummaryData | undefined> {
-        const summarizingEvent = PerformanceEvent.start(this.logger, {
-            eventName: "GenerateSummary",
-            message,
+        const summarizeEvent = PerformanceEvent.start(this.logger, {
+            eventName: "Summarize",
+            reason,
+            refreshLatestAck,
+            fullTree,
             timeSinceLastAttempt: Date.now() - this.heuristics.lastAttempted.summaryTime,
             timeSinceLastSummary: Date.now() - this.heuristics.lastAcked.summaryTime,
-            safe: safe || undefined,
         });
+        // Helper function to report failures and return.
+        const fail = <K extends keyof ISummarizeError>(
+            message: K,
+            description: ISummarizeError[K],
```

Comment:
description strings will be present twice in the code, adding to bundle size. Can this function retrieve description from ISummarizeError? Maybe by changing ISummarizeError shape (it's essentially a map).

Reply:
Actually, can we get rid of strings (description) altogether? English strings are not useful anyway (other than for telemetry), clients are much better to work with IDs that they can map to their localized strings if needed. raiseSummarizingError() ideally should either have enum arg, or some kind of structured error (ideally relying on same structure as everywhere else - defining its on subset of errorType(s)

Reply:
`ISummarizeError` as an interface should only exists before being transpiled and bundled into js code. It will appear 2x in source code/typings only.

Do you think I should promote `ISummarizeError` to be a readonly const object: `summarizeErrors`, acting as a map from code to description?

Reply:
I'm not sure descriptions are valuable / useful at all. What's the scenario? They might be useful only in telemetry (as they are not localized), and I think code are as good there.

Reply:
Went ahead and converted `ISummarizeError` interface to `summarizeErrors` const object. We now use that to map from code/key to description.